server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -472,10 +474,79 @@ private void executeSearch(SearchTask task, SearchTimeProvider timeProvider, Sea
+                //that we don't keep the search context open for too long between query and fetch for ordinary indices due to slow indices.
+                CountDown countDown = new CountDown(2);
+                AtomicReference<Exception> exceptions = new AtomicReference<>();
+                SearchResponseMerger searchResponseMerger = createSearchResponseMerger(searchRequest.source(), timeProvider,
+                    searchService::createReduceContext);
+                CountDownActionListener<SearchResponse, SearchResponse> countDownActionListener =
+                    new CountDownActionListener<>(countDown, exceptions, listener) {
+                    @Override
+                    void innerOnResponse(SearchResponse searchResponse) {
+                        searchResponseMerger.add(searchResponse);
+                    }
+
+                    @Override
+                    SearchResponse createFinalResponse() {
+                        return searchResponseMerger.getMergedResponse(clusters);
+                    }
+                };
+
+                String[] ordinaryIndices = nonThrottledIndices.toArray(new String[0]);
+                //Note that the indices set to the new SearchRequest won't be retrieved from it, as they have been already resolved and
+                //will be provided separately to executeSearch.
+                SearchRequest nonThrottledRequest = SearchRequest.subSearchRequest(searchRequest, ordinaryIndices,
+                    RemoteClusterService.LOCAL_CLUSTER_GROUP_KEY, timeProvider.getAbsoluteStartMillis(), false);
+                executeSearch(task, timeProvider, nonThrottledRequest, localIndices, ordinaryIndices, routingMap,
+                    aliasFilter, concreteIndexBoosts, remoteShardIterators, remoteConnections, clusterState, countDownActionListener,
+                    SearchResponse.Clusters.EMPTY);
+
+                String[] throttled = throttledIndices.toArray(new String[0]);
+                //Note that the indices set to the new SearchRequest won't be retrieved from it, as they have been already resolved and
+                //will be provided separately to executeSearch.
+                SearchRequest throttledRequest = SearchRequest.subSearchRequest(searchRequest, throttled,
+                    RemoteClusterService.LOCAL_CLUSTER_GROUP_KEY, timeProvider.getAbsoluteStartMillis(), false);
+                //as we are searching throttled indices only, we can set this to 1 automatically so the can match phase is always run
+                throttledRequest.setPreFilterShardSize(1);
+                executeSearch(task, timeProvider, throttledRequest, localIndices, throttled, routingMap,
+                    aliasFilter, concreteIndexBoosts, Collections.emptyList(), (alias, id) -> null, clusterState, countDownActionListener,
+                    SearchResponse.Clusters.EMPTY);
```

Comment:
I was planning on adding tests to TransportSearchActionSingleNodeTests, but it looks like I can not make an index search throttled manually (it's a private setting), hence I am not sure how to search against throttled indices from there.

Reply:
I added some tests for this [here](https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/search/SearchServiceTests.java#L585)

Reply:
thanks for the pointer, I will look at that